### PR TITLE
Add zipaction integration test cases.

### PR DESCRIPTION
**Pull request by lzbj**
Address issue #245.

### CODE REVIEW

**Comment by lzbj** [comment]
@houshengbo , please help review, it addressed issue 245 as you opened, please notice maybe later we'll change the behavior, because now the user needs to install the deps manually, we plan to support the automatic install deps. As issue https://github.com/openwhisk/openwhisk-wskdeploy/issues/250 mentioned.

**Comment by houshengbo** [comment]
@lzbj Do we actually need the files under node_modules/left-pad? 
You submitted a lot of added packages, which makes me confused. Are they really necessary? This PR is supposed to fix the zip action, but where is the zip file?
I personally do not have a positive feeling about this left-pad package. There is even the "F" word inside some of the files.

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/.npmignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+npm-debug.log
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/.npmignore]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+ - "6"
+ - "5"
+ - "4"
+ - "0.12"
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/.travis.yml]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/COPYING [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+ Copyright (C) 2014 Azer Koçulu <azer@roadbeats.com>
+
+ Everyone is permitted to copy and distribute verbatim or modified
+ copies of this license document, and changing it is allowed as long
+ as the name is changed.
+
+            DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
+   TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
+
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/COPYING]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+## left-pad
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/README.md]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/* This program is free software. It comes without any warranty, to
+     * the extent permitted by applicable law. You can redistribute it
+     * and/or modify it under the terms of the Do What The Fuck You Want
+     * To Public License, Version 2, as published by Sam Hocevar. See
+     * http://www.wtfpl.net/ for more details. */
+'use strict';
+module.exports = leftPad;
+
+var cache = [
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/index.js]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/package.json [hunk 6]
```diff
@@ -0,0 +1,103 @@
+{
+  "_args": [
+    [
+      {
+        "raw": "left-pad@1.1.3",
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/package.json]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/O(n).js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+module.exports = function (str, len, ch) {
+  str = str + '';
+
+  len = len - str.length;
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/O(n).js]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/es6Repeat.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = function (str, len, ch) {
+  str = str + '';
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/es6Repeat.js]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/perf.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+var oN = require('./O(n)');
+var es6Repeat = require('./es6Repeat');
+var current = require('../');
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/perf.js]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/node_modules/left-pad/test.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
+/* This program is free software. It comes without any warranty, to
+     * the extent permitted by applicable law. You can redistribute it
+     * and/or modify it under the terms of the Do What The Fuck You Want
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/node_modules/left-pad/test.js]: Do we need this file?

File: tests/src/integration/zipaction/actions/padding/package.json [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "my-action",
+  "main": "index.js",
```
Comment (houshengbo) [inline tests/src/integration/zipaction/actions/padding/package.json]: Do we need this file?

Comment (lzbj) [inline tests/src/integration/zipaction/actions/padding/package.json]: @houshengbo , I reply all your comments here: this is a sample use case from the openwhisk zip action doc, for zipped actions, yes, that's necessary, you can try by running npm install under the action.

**Comment by lzbj** [comment]
@houshengbo @mrutkows , could you please have a review, I updated and removed some unnecessary files, except ones that sketch the node module hierarchy, sorry I didn't noticed there are some bad words, I deleted the file.

**Comment by houshengbo** [comment]
@lzbj The package you are using is not appropriate, since the license of this package is weird and has inappropriate language in it. I suggest you switch to use another package.

**Comment by lzbj** [comment]
@houshengbo , thanks for review, yes, I'll find another package with minimum footprint.

**Comment by lzbj** [comment]
@houshengbo , I updated the referenced package and removed the unnecessary files, please help review.

**Comment by lzbj** [comment]
@mrutkows @daisy-ycguo , could you please help review this pr, it seems Vincent is busy on other projects.

File: tests/src/integration/zipaction/deployment.yml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+application:
+  name: wskdeploy-samples
+  namespace: guest
+  credential: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502:123zO3xZCLrMN6v2BKK1dXYFpXlPkccOFqm12CdAsMgRU4VrNZ9lyGVCGuMDGIwP
```
Comment (mrutkows) [inline tests/src/integration/zipaction/deployment.yml]: @houshengbo is including a key here correct?

File: tests/src/integration/zipaction/zipaction_test.go [hunk 13]
```diff
@@ -22,6 +23,6 @@ func TestZipAction(t *testing.T) {
 }
 
 var (
-	manifestPath   = os.Getenv("GOPATH") + "/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/zipaction/manifest.yaml"
-	deploymentPath = "" 
+	manifestPath   = os.Getenv("GOPATH") + "/src/github.com/openwhisk/openwhisk-wskdeploy/tests/src/integration/zipaction/manifest.yml"
+	deploymentPath = os.Getenv("GOPATH") + "/src/github.com/openwhisk/openwhisk-wskdeploy/tests/src/integration/zipaction/deployment.yml"
```
Comment (mrutkows) [inline tests/src/integration/zipaction/zipaction_test.go]: The path should reflect the new "apache" organization.

File: utils/misc.go [hunk 14]
```diff
@@ -192,6 +192,10 @@ func CreateFolderZip(src, des string) error {
 		writer, err := zipWritter.CreateHeader(header)
```
Comment (mrutkows) [inline utils/misc.go]: Please correct object name to "ZipWriter" (from "ZipWritter")

File: tests/src/integration/zipaction/manifest.yml [hunk 15]
```diff
@@ -0,0 +1,15 @@
+package:
+  name: zipaction
+  version: 1.0
+  license: Apache-2.0
+  actions:
+    cat:
+      version: 1.0
+      location: actions/cat
```
Comment (mrutkows) [inline tests/src/integration/zipaction/manifest.yml]: This is a good simple (in-line), single action test (better than previous one).  We should perhaps create another test that uses multiple (sub)directories.

File: utils/misc.go (hunk 14 above)
Comment (lzbj) [inline utils/misc.go]: Ok, sure.

File: tests/src/integration/zipaction/manifest.yml (hunk 15 above)
Comment (lzbj) [inline tests/src/integration/zipaction/manifest.yml]: Ok, I'll open another issue to track it.

File: tests/src/integration/zipaction/zipaction_test.go (hunk 13 above)
Comment (lzbj) [inline tests/src/integration/zipaction/zipaction_test.go]: Yes, correct.

File: utils/misc.go [hunk 16]
```diff
@@ -156,96 +156,59 @@ func GetJSONType(j interface{}) string {
...
-		Check(err)
-
-		file, err := os.Open(path)
-		Check(err)
-		defer file.Close()
-		_, err = io.Copy(writer, file)
-		Check(err)
-		return err
-	})
+func NewZipWritter(src, des string) *ZipWritter {
```
Comment (mrutkows) [inline utils/misc.go]: Should be "ZipWriter"

File: utils/misc.go [hunk 17]
```diff
@@ -156,96 +156,59 @@ func GetJSONType(j interface{}) string {
...
+	zw := &ZipWritter{src: src, des: des}
+	return zw
+}
 
-	return err
+type ZipWritter struct {
```
Comment (mrutkows) [inline utils/misc.go]: name should be "ZipWriter"

File: tests/src/integration/zipaction/actions/cat/index.js [hunk 18]
```diff
@@ -0,0 +1,5 @@
+function myAction(args) {
+    var cat = require('cat');
+    cat('https://baidu.com', console.log);
```
Comment (mrutkows) [inline tests/src/integration/zipaction/actions/cat/index.js]: Change this to a neutral website (like openwhisk.org) <or even better> do not require HTTPS and "cat" some local file?

**Comment by mrutkows** [comment]
@houshengbo the new testcase looks good (simple

**Comment by pritidesai** [comment]
Thanks a bunch David for this fix, I downloaded source from this pull request, built wskdeploy binary and tested it. I was able to create `cat-get-action` action as stated in #269 and run it successfully:

```
wsk action invoke cat-get/cat-get-action --blocking --param id 2
"response": {
        "result": {
            "color": "red",
            "id": 2,
            "name": "minnie"
        },
        "status": "success",
        "success": true
    },
```

**Pull request merged**